### PR TITLE
Added AtEndOfDay method to date extensions. Added Tests for AtEndOfDa…

### PR DESCRIPTION
**Pull request by saidshahdev**
A AtEndOfDay method was added to DateExtensions to allow a DateTimeOffsets time to be set to 11:59:59 pm. A Rider specific section was added to the gitIgnore file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,6 +49,9 @@ bld/
 # Visual Studio 2017 auto generated files
 Generated\ Files/
 
+# Rider
```
Comment (wintondeshong) [inline .gitignore:52]: Please move any 'custom' additions near the top (above line 13). At around line 13 we are leveraging the microsoft gitignore.

```
## Below Ignores Visual Studio temporary files, build results, and
## files generated by popular Visual Studio add-ons. Copied directly
## from https://github.com/github/gitignore/blob/master/VisualStudio.gitignore
```

File: src/AndcultureCode.CSharp.Extensions/DateExtensions.cs [hunk 2]
```diff
@@ -6,6 +6,23 @@ namespace AndcultureCode.CSharp.Extensions
...
     public static class DateExtensions
     {
+        /// <summary>
+        /// Sets and returns the time to 11:59:59 on the given DateTimeOffset.
+        /// </summary>
+        /// <param name="date"></param>
+        /// <returns></returns>
+        public static DateTimeOffset AtEndOfDay(this DateTimeOffset date)
+        {
+            return new DateTimeOffset(
```
Comment (wintondeshong) [inline src/AndcultureCode.CSharp.Extensions/DateExtensions.cs:16]: Refactoring food for thought: Consider expression bodies.

```csharp
public static DateTimeOffset AtEndOfDay(this DateTimeOffset date)
    => new DateTimeOffset(...); 
```

File: test/AndcultureCode.CSharp.Extensions.Tests/DateExtensionsTests.cs [hunk 3]
```diff
@@ -9,6 +9,54 @@ namespace AndcultureCode.CSharp.Extensions.Tests
...
+            var testDate = DateTimeOffset.Now;
+
+            // Act
+            var result = testDate.AtEndOfDay();
+
+            // Assert
+            result.Hour.ShouldBe(23);
+            result.Minute.ShouldBe(59);
+            result.Second.ShouldBe(59);
+
```
Comment (wintondeshong) [inline test/AndcultureCode.CSharp.Extensions.Tests/DateExtensionsTests.cs:27]: Minor, extra newline

File: .gitignore (hunk 1 above)
Comment (saidshahdev) [reply .gitignore]: @wintondeshong The changes have been made. Can you re-review when you get a chance?

**Comment by wintondeshong** [comment]
@all-contributors please add @ghost for code and tests

**Pull request merged**